### ISSUE TITLE
Transpiling 2i to 2.i

### ISSUE THREAD

**Issue by eregon** [ruby-next/ruby-next#67]
I'm trying to run some small scripts on Ruby 2.0.0 (to compare performance), and one issue I'm seeing is that something like `1 + 2i` doesn't parse on 2.0.
The `2i` notation was added in Ruby 2.1.

Do you think this is something that would make sense to support in ruby-next?

Note that I don't request support for Ruby 2.0, just transpiling `2i` to `2.i`.
I actually got some script using `case in` working on 2.0 with:
```
ruby-next core_ext -o ruby-next.rb
ruby-next -V --min-version=2.0 nextify script.rb
ruby -r./backport.rb -r./ruby-next.rb .rbnext/2.7/script.rb
```

**Comment by palkan** [comment]
> Do you think this is something that would make sense to support in ruby-next?

I think, if it could be useful for someone (even a single person 😉) that makes sense. Especially, since the effort to implement it doesn't seem too big.

Do we expect the following:

```ruby
1 + 2i #=> 1 + Complex(0, 1)

# also, rationals
1/2r #=> 1 / Rational(2)
```

?

**Comment by eregon** [comment]
There is a `.i` method, so I think that might be simpler and also more efficient:
In Ruby 2.0: `1 + 2.i # => (1+2i)`.
```
$ ruby -rbenchmark -e 'p Benchmark.realtime { i=0; while i<10_000_000; Complex(0, 2); i+=1; end } '
2.013413995
$ ruby -rbenchmark -e 'p Benchmark.realtime { i=0; while i<10_000_000; 2.i; i+=1; end } '          
0.890864725
```

For the `2r` notation, there is no `.r` though, so indeed that needs `.to_r` or `Rational()`.

**Comment by palkan** [comment]
Released in 0.12.0 (along with &. fixes and some other Ruby 2.0 stuff).
